workspace/__lib__/xod/json/parse-json/patch.cpp

```diff
@@ -0,0 +1,446 @@
+
+using xod::xod__json__parsed_json_stream::ParserState;
+
+#define STACK_OBJECT             0
+#define STACK_ARRAY              1
+#define STACK_KEY                2
+#define STACK_STRING             3
+
+bool isFirstCharInNumber(char c) {
+    return (c >= '0' && c <= '9') || c == '-';
+}
+
+// We don't use `strchr` because buffer we search in is not \0 - terminated
+bool doesCharArrayContain(char myArray[], int length, char c) {
+    for (int i = 0; i < length; i++) {
+      if (myArray[i] == c) {
+        return true;
+      }
+    }
+    return false;
+}
+
+struct ContextObject;
+
+// we only use buffer to validate numbers, booleans and `null`
+#define BUFFER_MAX_LENGTH 64
```

Comment:
In modern C++ prefer constexpr

```suggestion
constexpr size_t BUFFER_MAX_LENGTH = 64;
```

And are we really need to keep 64 bytes? The number we can precisely keep is ~7 digits. A 15-digit number is huge. I think we could limit ourselves to 16 chars.